### PR TITLE
Kubernetes events

### PR DESCRIPTION
**Pull request by aa1ex**

### CODE REVIEW

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -4,5 +4,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: example.com/vector-operator
-  newTag: v0.0.1
+  newName: localhost:5001/vector-operator
+  newTag: dev06
```
Comment (zvlb) [inline config/manager/kustomization.yaml:8]: rollback

File: cmd/main.go [hunk 2]
```diff
@@ -208,13 +209,16 @@ func main() {
 	vectorAggregatorsPipelineEventCh := make(chan event.GenericEvent, 10)
 	defer close(vectorAggregatorsPipelineEventCh)
 
+	evCollector := k8sevents.NewEventsCollector(clientset, ctrl.Log.WithName("kubernetes-events-manager"))
```
Comment (zvlb) [inline cmd/main.go:212]: kubernetes-events-collector

File: internal/vector/aggregator/service.go [hunk 3]
```diff
@@ -29,8 +31,19 @@ func (ctrl *Controller) ensureVectorAggregatorService(ctx context.Context) error
...
+			ctrl.EventsCollector.RegisterSubscriber(
+				svc.Name,
+				svc.Namespace,
+				svc.Annotations[common.AnnotationK8sEventsPort],
+				svc.Annotations[common.AnnotationK8sEventsNamespace],
+			)
+		}
 	}
 	for _, svc := range existing {
+		if svc.Annotations[common.AnnotationK8sEventsNamespace] != "" {
```
Comment (zvlb) [inline internal/vector/aggregator/service.go:44]: add port

**Review by zvlb — COMMENTED** [review]
need small fixes

File: internal/controller/service_controller.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if len(svc.Spec.Ports) == 1 {
+		namespace, _ := svc.Annotations[common.AnnotationK8sEventsNamespace]
+		host := fmt.Sprintf("%s.%s", svc.Name, svc.Namespace)
+		port := strconv.Itoa(int(svc.Spec.Ports[0].Port))
+		protocol := strings.ToLower(string(svc.Spec.Ports[0].Protocol))
+		r.EventsManager.RegisterSubscriber(req.NamespacedName.String(), host, port, protocol, namespace)
+	}
```
[the remark this answers is not in the document]
Comment (aa1ex) [reply internal/controller/service_controller.go]: outdated

File: cmd/main.go (hunk 2 above)
Comment (aa1ex) [reply cmd/main.go:212]: fixed

File: internal/vector/aggregator/service.go (hunk 3 above)
Comment (aa1ex) [reply internal/vector/aggregator/service.go:44]: fixed

File: config/manager/kustomization.yaml (hunk 1 above)
Comment (aa1ex) [reply config/manager/kustomization.yaml:8]: done.

**Review by zvlb — APPROVED** [review]
LGTM

**Pull request merged**